mlflow/xgboost.py

```diff
@@ -416,10 +437,36 @@ def log_feature_importance_plot(features, importance, importance_type):
             finally:
                 shutil.rmtree(tmpdir)
 
-        try_mlflow_log(log_model, model, artifact_path="model")
+        # dtrain must exist as the original train function already ran successfully
+        dtrain = args[1] if len(args) > 1 else kwargs["dtrain"]
+
+        input_example = None
+        signature = None
+        try:
+            input_example = dtrain.data_copy
+
+            # it is possible that the dataset was constructed before the patched
+            #   constructor was applied, so we cannot assume the data_copy exists
+            if input_example is None:
```

Comment:
In the case where the dataset was constructed prior to patching, won't we fail at the first line in this block (`input_example = dtrain.data_copy`) with an `AttributeError`? Instead, should we use `getattr()` to access the `data_copy` attribute to ensure that accessing the attribute doesn't throw?

Reply:
ah, that's correct. technically the autologging will still continue because it's in a try block, but I definitely wanted it to take the `if input_example is None` path instead